core/include/moveit/task_constructor/moveit_compat.h

```diff
@@ -57,3 +58,10 @@
 #define MOVEIT_HAS_OBJECT_POSE MOVEIT_VERSION_GE(1, 1, 6)
 
 #define MOVEIT_HAS_STATE_RIGID_PARENT_LINK MOVEIT_VERSION_GE(1, 1, 6)
+
+#if !MOVEIT_VERSION_GE(1, 1, 9)
+// the pointers are not yet available
+namespace trajectory_processing {
+MOVEIT_CLASS_FORWARD(TimeParameterization);
+}
+#endif
```

Comment:
I just saw you [backported the polymorphic base class change to melodic](https://github.com/ros-planning/moveit/commit/393815e4872de6349ae3784360dbe276a9abda7a). So this should be unnecessary.
But Jafar and you both did not notice the missing pointers in the original patch, so I just filed [this PR](https://github.com/ros-planning/moveit/pull/3078). It might be worth creating a patch release for melodic with this addition as well. If you should do that, these lines can go away
```suggestion
```

Reply:
Until we have a new release of MoveIt, this addition is required. Do you propose a patch release of MoveIt before? 
I could do that over the weekend indeed...

Reply:
I guess you can remove this now. :)